extensions/openai-translator/src/providers/prompt.ts

```diff
@@ -0,0 +1,245 @@
+          result = this.prevQuoteEndBuffer + textDelta;
+          quoteEndBuffer = "";
+          break;
+        }
+      }
+    }
+    // console.debug('end loop!')
+    this.prevQuoteEndBuffer = quoteEndBuffer;
+    // console.debug('totally result', result)
+    // console.debug('this.quoteEndBuffer', this.quoteEndBuffer)
+    // console.debug('---- end of process quote end -----')
+    return result;
+  }
+}
+
+export function generatMetadata(query: TranslateQuery): PromptMetadata {
+  const sourceLangCode = query.detectFrom;
+  const targetLangCode = query.detectTo;
+  const sourceLang = lang.getLangName(sourceLangCode);
+  const targetLang = lang.getLangName(targetLangCode);
+  console.debug("sourceLang", sourceLang);
+  console.debug("targetLang", targetLang);
+  const toChinese = lang.isChineseLangCode(targetLangCode);
+  const isWordMode = toChinese && isAWord(sourceLangCode, query.text.trim());
+
+  return {
+    content: query.text,
+    sourceLangCode,
+    targetLangCode,
+    sourceLang,
+    targetLang,
+    isWordMode,
+    toChinese,
+  };
+}
+
+export const promptBuilders: Record<TranslateMode, PromptBuilder> = {
+  // Translate
+  ["translate"]: (prompt: Prompt) => {
+    let { rolePrompt, commandPrompt, contentPrompt } = prompt;
+    const { quoteProcessor = new QuoteProcessor(), meta: query } = prompt;
+    const { content, isWordMode, targetLangCode, targetLang, toChinese } = query;
+    if (isWordMode) {
+      // 翻译为中文时，增加单词模式，可以更详细的翻译结果，包括：音标、词性、含义、双语示例。
+      rolePrompt = `你是一个翻译引擎，请将翻译给到的文本，只需要翻译不需要解释。当且仅当文本只有一个单词时，请给出单词原始形态（如果有）、单词的语种、对应的音标（如果有）、所有含义（含词性）、双语示例，至少三条例句，请严格按照下面格式给到翻译结果：
+                <单词>
+                [<语种>] · / <单词音标>
+                [<词性缩写>] <中文含义>]
+                例句：
+                <序号><例句>(例句翻译)`;
+      commandPrompt = "好的，我明白了，请给我这个单词。";
+      contentPrompt = `单词是：${content}`;
+      return { ...prompt, rolePrompt, commandPrompt, contentPrompt };
+    } else {
+      commandPrompt += ` Only translate the text between ${quoteProcessor.quoteStart} and ${quoteProcessor.quoteEnd}.`;
+      contentPrompt = `${quoteProcessor.quoteStart}${content}${quoteProcessor.quoteEnd} =>`;
+      if (targetLangCode === "xdbhw") {
+        rolePrompt = "您是一位在中文系研究中文的资深学者";
+        commandPrompt = `夹在${quoteProcessor.quoteStart}和${quoteProcessor.quoteEnd}之间的内容是原文，请您将原文内容翻译成《呐喊》风格的现代白话文`;
+      } else if (targetLangCode === "jdbhw") {
+        rolePrompt = "您是一位在中文系研究中文的资深学者";
+        commandPrompt = `夹在${quoteProcessor.quoteStart}和${quoteProcessor.quoteEnd}之间的内容是原文，请您将原文内容翻译成《红楼梦》风格的近代白话文`;
+      } else if (content.length < 5 && toChinese) {
+        // 当用户的默认语言为中文时，查询中文词组（不超过5个字），展示多种翻译结果，并阐述适用语境。
+        rolePrompt = `你是一个翻译引擎，请将给到的文本翻译成${targetLang}。请列出3种（如果有）最常用翻译结果：单词或短语，并列出对应的适用语境（用中文阐述）、音标、词性、双语示例。按照下面格式用中文阐述：
+                    <序号><单词或短语> · /<音标>
+                    [<词性缩写>] <适用语境（用中文阐述）>
+                    例句：<例句>(例句翻译)`;
+        commandPrompt = "";
+      }
+    }
```

Comment:
At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension.

Reply:
Hi @pernielsentikaer,  This is not a localization . Instead, it is a prompt to optimize the quality of Chinese translation.  It is invisible to users.